sklearn/impute.py

```diff
@@ -422,6 +423,25 @@ def transform(self, X):
 
         return X
 
+    def get_feature_names(self, input_features=None):
+        """Get feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of string
+            Input feature names.
+
+        Returns
+        -------
+        feature_names : array-like of string
+            Transformed feature names
+        """
+        check_is_fitted(self, 'statistics_')
+        if input_features is None:
+            input_features = ['x%d' % i
```

Comment:
shouldn't these check if they have `input_features_` set and use that instead?

Reply:
good question. Probably. Maybe let's discuss in the slep?